ledger/blockHeaderCache.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package ledger
+
+import (
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+
+	"github.com/algorand/go-deadlock"
+)
+
+const latestHeaderCacheSize = 512
+const blockHeadersLRUCacheSize = 10
+
+// blockHeaderCache is a wrapper for all block header cache mechanisms used within the Ledger.
+type blockHeaderCache struct {
```

Comment:
so why a new block header cache in front of the block cache of the last 512 blocks? also, there is another header cache in txtail now of the last 1001 blocks. plus the old heapLRUCache.

Reply:
it is definitely doable to remove this cache. However, it might take some time because we need to tweak the txtail so it can be extended over 1000 rounds back if needed (as the block queue currently does).

Reply:
committedUpTo  sets retain length to whatever voters needs, so block queue should have all required blocks (not cached tho)